core/src/main/java/org/infinispan/encoding/DataConversion.java

```diff
@@ -166,28 +173,17 @@ public void injectDependencies(@ComponentName(KnownComponentNames.CACHE_NAME) St
       }
       boolean internalCache = icr.isInternalCache(cacheName);
       boolean embeddedMode = Configurations.isEmbeddedMode(gcr);
-      this.storageMediaType = getStorageMediaType(configuration, embeddedMode, internalCache);
+      this.storageMediaType = getStorageMediaType(configuration, embeddedMode, internalCache, persistenceMarshaller);
```

Comment:
This was added to avoid requiring caches to configure a storage media type, and to assume sensible defaults. We can discuss if this config should be mandatory or not, but this is out of scope for this PR

Reply:
IIUC `injectDependencies()` is only called on the remote nodes, not on the origin node. But `storageMediaType` is not transient, so the value coming from the origin node should be good enough. Or if `injectDependencies()` is also called on the origin node, then we don't need to initialize it in the constructor.

Or were you thinking about a cache that is configured with one storage media type on node A and another storage media type on node B? I'm pretty sure that would break lots of other things, because we don't transcode values in non-functional commands.

Reply:
Here's where the DataConversion is injected: https://github.com/infinispan/infinispan/blob/db04d4aa1c19775dbb364daf7c3a7b9eb36ec514/core/src/main/java/org/infinispan/cache/impl/EncoderCache.java#L122-L125

Reply:
I was wrong, `injectDependencies()` is called, it just skips most of the initialization, including that of `storageMediaType` (see line 167)

Reply:
One more thing: since we only wire the dependency into `DataConversion` instances, but we never register them in the component registry, I suggest changing `@Scope(NAMED_CACHE)` to `@Scope(NONE)`.

Reply:
Moving the ```this.encoderRegistry = encoderRegistry``` below the ```if``` breaks a bunch of stuff, since the registry can be used in ```convertToRequestFormat()```

Reply:
@gustavonalle the if should only apply to the static instances like `DataConversion.DEFAULT_KEY`. The encoder registry is a component, so 2 nodes running in the same JVM will overwrite each other's `encoderRegistry` reference, which I don't think is correct.

Reply:
AFAICT, the same encoder registry instance is inject in all data conversions of all caches, it's a cache manager level component. If you have 2 nodes in the same JVM, I supposed each node will have a different one, but they won't mix?

Reply:
Actually, I won't do this for this PR, as it's unrelated. 
https://issues.redhat.com/browse/ISPN-11620

Reply:
Excellent, I was actually testing my changes locally and came to the same conclusion: a `DataConversion` instance should only convert between the storage type and the request type, the transcoder lookup should be elsewhere.